src/java.security.jgss/share/classes/sun/security/krb5/internal/crypto/EType.java

```diff
@@ -48,17 +48,46 @@
 public abstract class EType {
 
     private static final boolean DEBUG = Krb5.DEBUG;
-    private static boolean allowWeakCrypto;
+
+    // etypes supported by JDK, including weak ones
+    private static int[] supportedETypes;
+    // common default etypes if not defined in krb5.conf
+    private static int[] defaultETypes;
 
     static {
         initStatic();
     }
 
     public static void initStatic() {
-        boolean allowed = false;
+
+        // There are at most 8 etypes.
+        // Note: add etypes in preference order, because the default enctypes
+        // list is also a part of it.
+        int[] result = new int[8];
+        int num = 0;
+
+        int maxKeyLength = 0;
+        try {
+            maxKeyLength = Cipher.getMaxAllowedKeyLength("AES");
+        } catch (Exception e) {
+            // should not happen
+        }
+        if (maxKeyLength >= 256) {
+            result[num++] = EncryptedData.ETYPE_AES256_CTS_HMAC_SHA1_96;
+        }
+        result[num++] = EncryptedData.ETYPE_AES128_CTS_HMAC_SHA1_96;
+        if (maxKeyLength >= 256) {
+            result[num++] = EncryptedData.ETYPE_AES256_CTS_HMAC_SHA384_192;
+        }
+        result[num++] = EncryptedData.ETYPE_AES128_CTS_HMAC_SHA256_128;
+
+        // By default, only AES etypes are enabled
+        defaultETypes = Arrays.copyOf(result, num);
+
+        boolean allowWeakCrypto = false;
         try {
             Config cfg = Config.getInstance();
-            allowed = cfg.getBooleanObject("libdefaults", "allow_weak_crypto")
+            allowWeakCrypto = cfg.getBooleanObject("libdefaults", "allow_weak_crypto")
```

Comment:
nit: exceeds 80 chars?

Reply:
A little. I've learnt to accept lines a little bit longer than 80.